BUILD

```diff
@@ -0,0 +1,35 @@
+cc_library(
+  name = "marl",
+  includes = [
+    "include",
+  ],
+  hdrs = glob([
+    "include/marl/**/*.h",
+  ]),
+  srcs = glob([
+    "src/**/*.cpp",
+    "src/**/*.c",
+    "src/**/*.S",
+    "src/**/*.h",
+  ], exclude = glob([
+    "src/**/*_test.cpp",
+  ])),
+  linkopts = select({
+    "@bazel_tools//src/conditions:linux_x86_64": [ "-pthread" ],
```

Comment:
@ben-clayton what other platforms will need libpthread? I'm assuming at least the Android ones will need it as well.

Reply:
Likely, yes, but I'm happy to evolve the build rules than try to get them all perfect first time.